pkg/commands/compute/language_toolchain.go

```diff
@@ -179,6 +196,55 @@ func (bt BuildToolchain) Build() error {
 	return nil
 }
 
+// The encoding of a module starts with a preamble containing a 4-byte magic
+// number (the string '\0asm') and a version field.
+//
+// Reference:
+// https://webassembly.github.io/spec/core/binary/modules.html#binary-module
+func (bt BuildToolchain) validateWasm(path string) error {
+	// gosec flagged this:
+	// G304 (CWE-22): Potential file inclusion via variable
+	// Disabling as we trust the source of the path variable.
+	// #nosec
+	f, err := os.Open(path)
+	if err != nil {
+		return bt.handleError(err)
+	}
+	defer f.Close()
+
+	// Parse the magic number
+	magic := make([]byte, wasmBytes)
+	_, err = f.Read(magic)
+	if err != nil {
+		return bt.handleError(err)
+	}
+	if len(magic) > 0 && magic[0] == 0x00 {
+		magic = magic[1:] // Remove the first byte which is a null character "\x00"
+	}
+	if string(magic) != "asm" {
+		return bt.handleError(err)
+	}
+	if bt.verbose {
+		text.Break(bt.out)
+		text.Description(bt.out, "Wasm module 'magic'", string(magic))
+	}
+
+	// Parse the version
+	versionBytes := make([]byte, wasmBytes)
+	_, err = f.Read(versionBytes)
+	if err != nil {
+		return bt.handleError(err)
+	}
+	version := int(binary.LittleEndian.Uint32(versionBytes))
+	if version <= 0 {
+		return bt.handleError(err)
+	}
```

Comment:
This is fine, but you can also do

```go
var version uint32
if err := binary.Read(f, binary.LittleEndian, &version); err != nil {
	return bt.handleError(err)
}
```

to save some lines.

Reply:
oh nice! thanks